formwidgets/MLNestedForm.php

```diff
@@ -0,0 +1,206 @@
+    }
+
+    /**
+     * Returns an array of translated values for this field
+     * @return array
+     */
+    public function getSaveValue($value)
+    {
+        $this->rewritePostValues();
+
+        return $this->getLocaleSaveValue(is_array($value) ? array_values($value) : $value);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function loadAssets()
+    {
+        $this->actAsParent();
+        parent::loadAssets();
+        $this->actAsParent(false);
+
+        if (Locale::isAvailable()) {
+            $this->loadLocaleAssets();
+            $this->addJs('js/mlnestedform.js');
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getParentViewPath()
+    {
+        return base_path().'/modules/backend/formwidgets/nestedform/partials';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getParentAssetPath()
+    {
+        return '/modules/backend/formwidgets/nestedform/assets';
+    }
+
+    public function onSwitchItemLocale()
+    {
+        if (!$locale = post('_nestedform_locale')) {
+            throw new ApplicationException('Unable to find a nestedform locale for: '.$locale);
+        }
+
+        // Store previous value
+        $previousLocale = post('_nestedform_previous_locale');
+        $previousValue = $this->getPrimarySaveDataAsArray();
+
+        // Update widget to show form for switched locale
+        $lockerData = $this->getLocaleSaveDataAsArray($locale) ?: [];
+        $this->reprocessLocaleItems($lockerData);
+
+        foreach ($this->formWidgets as $key => $widget) {
+            $value = array_shift($lockerData);
+            if (!$value) {
+                unset($this->formWidgets[$key]);
+            }
+            else {
+                $widget->setFormValues($value);
+            }
+        }
+
+        $this->actAsParent();
+        $parentContent = parent::render();
+        $this->actAsParent(false);
+
+        return [
+            '#'.$this->getId('mlNestedForm') => $parentContent,
+            'updateValue' => json_encode($previousValue),
+            'updateLocale' => $previousLocale,
+        ];
+    }
+
+    /**
+     * Ensure that the current locale data is processed by the repeater instead of the original non-translated data
+     * @return void
+     */
+    protected function reprocessLocaleItems($data)
+    {
+        $this->formWidgets = [];
+        $this->formField->value = $data;
+
+        $key = implode('.', HtmlHelper::nameToArray($this->formField->getName()));
+        $requestData = Request::all();
+        array_set($requestData, $key, $data);
+        Request::merge($requestData);
+
+        $this->processItems();
+    }
+
+    /**
+     * Gets the active values from the selected locale.
+     * @return array
+     */
+    protected function getPrimarySaveDataAsArray()
+    {
+        $data = post($this->formField->getName()) ?: [];
+
+        return $this->processSaveValue($data);
```

Comment:
```suggestion
        return $data;
```

Reply:
@iboved can you try this change to see if that works?

Reply:
@mjauvin now the error is that `processItems()` method does not exist. Called [here](https://github.com/wintercms/wn-translate-plugin/pull/9/files#diff-c2f342249b41720fcef2b17f86801e6006272c04ee121174aa91e542a7f885d5R148).